setup postgres database connection with node js ecommerce app

BenoitIT opened the pull request:
## Postgres  database setup
#### set up connection between node js application and postgres database so that app can access and send data to postgress database

### What does this PR do?
This has functions and methods to set up connection  [ecommerce-app-legends-bn](https://github.com/atlp-rwanda/ecommerce-app-legends-bn) App with postgres database during development phase.

#### what are the nature of PL contents?
---config folder which has js file with db connection configurations
---models folder which has User model created for functionality testing as well index.js which has db connection with 
     models
---migration folder which store table structures to be migrated to the database
---scripts to run both create migration events as well as make: db migrate
---connection folder which has folder to notify the the status of connection from the node js console

N.B : migrations and models folder are still in progress as new and more migrations and modls will b created

### Description of Tasks completed
----database connection set-up successfully
----first model created successfully
----first migration created and migrated successfully
--- make model and create migration as well as migrate scripts added and tested
### how to test this?
clone repo , branch git clone https://github.com/atlp-rwanda/ecommerce-app-legends-bn.git
-`run npm i`
-have `postgres database` installed 
-create a database with `ecommerce_db`as its name
-create .env file
-under .env file initialize:`DEV_DATABASE_URL=postgres://<username>:<password>@127.0.0.1:5432/ecommerce_db`
###### remember to take away angled bracket  beside username and password as well colon between those two from connection string above
-start the app and check connectivity in development `npm run dev `
-  run `npm run make:migration`  in terminal to make migation  to create new migration
-rename created migration file with your desired file
- run`npm run db:migrate` to create new table in database
-  run`npm run generate:seed` to create new database seed 
- run`npm run db:seed` to seed the database
-  run`npm run migrate:undo` to revert migrations
<img width="477" alt="Screen Shot 2023-03-02 at 10 57 22" src="https://user-images.githubusercontent.com/103331579/222380479-0ea90ced-3909-429b-a3a5-5da2b9f61f54.png">
 database seed

BenoitIT committed:
chore(setup postgres database)

## description
-install sequolize and sequolize-cli
-install pg and pg-hstore
-create database directory
-add config and migration directory
-add model and seed directory
-create connection method and generate connection string
-create the first migration
-create the first demo model and seeds
-add script to run model and migration creation
-add script to create seed
- add script to seed database as well as migratting database table
##finish database setup

mansurissa commented:
Put `database` folder in `src`

mansurissa commented on package.json:
```diff
@@ -7,6 +7,11 @@
     "build": "babel src -d dist",
     "start": "nodemon dist/index.js",
     "dev": "nodemon --exec babel-node src/index.js",
+    "migrate": "npx sequelize-cli db:migrate",
```
I would suggest before migrating and seeding to first undo migrations and seeds

mansurissa commented on src/index.js:
```diff
@@ -1,6 +1,7 @@
 import express from 'express';
 import rootRouter from './api/root';
 import docs from './docs/index';
+import { testConnection } from '../database/models';
```
`testConnection` sounds like you are testing something. a name like `databaseConnection` would be better

mansurissa reviewed (changes requested):
Work on the comments, requested changes and ignore `.DS_store`

Aimeana100 reviewed (changes requested):
Thank you  @BenoitIT, you have made really good job. 
 I reviwed your work, and i foud that you may have been sone some `mistakes` you have made.
- ` newschema--attributes ` in  ` "make:migration":"npx sequelize-cli model:generate --name newschema--attributes firstName:string,lastName:string,email:string",` should be space separate
- * Again  update your pull request message* approprietry `ex. ` : `npm run db:migrate` doesn't make sence.
-  Lastly i want to suggest that you follow the file structure:
--   Letting Database be in src folder shold be much better.
--  even if it is sequelize-cli template i would like to advice you, to also take `models` folder direct under src *i.e* : `src/models` and update `.eslintrc.js`  accordingly.

Aimeana100 commented:
> * npm run db:seed


> Work on the comments, requested changes and ignore `.DS_store`

and @BenoitIT  may ignore `package-lock.json` as someone has suggested previously.

BenoitIT commented:
thanks for feedback every mentioned response is on the way getting handled!

BenoitIT committed:
chore(database setup)
-removing database folder from root folder
-adding database folder from src folder
-refactring script from package.json
-adding description .env example

BenoitIT committed:
gitignore changes
